frontend/src/metabase/components/FieldValuesWidget.jsx

```diff
@@ -500,22 +508,30 @@ function isExtensionOfPreviousSearch(value, lastValue, options, maxResults) {
   );
 }
 
-function isSearchable(fields, disableSearch, disablePKRemappingForSearch) {
+export function isSearchable({
+  fields,
+  disableSearch,
+  disablePKRemappingForSearch,
+  valuesMode,
+}) {
   return (
     !disableSearch &&
     // search is available if:
     // all fields have a valid search field
-    fields.every(field => searchField(field, disablePKRemappingForSearch)) &&
-    // at least one field is set to display as "search" or is a list field that has an incomplete value set
-    fields.some(
-      f =>
-        f.has_field_values === "search" ||
-        (f.has_field_values === "list" && f.has_more_values === true),
-    ) &&
-    // and all fields are either "search" or "list"
-    fields.every(
-      f => f.has_field_values === "search" || f.has_field_values === "list",
-    )
+    // the component has already been set to "search" mode
+    // (`valuesMode` can be set by the dashboard parameter values endpoint call)
+    (valuesMode === "search" ||
+      (fields.every(field => searchField(field, disablePKRemappingForSearch)) &&
+        // at least one field is set to display as "search" or is a list field that has an incomplete value set
+        fields.some(
+          f =>
+            f.has_field_values === "search" ||
+            (f.has_field_values === "list" && f.has_more_values === true),
+        ) &&
+        // and all fields are either "search" or "list"
+        fields.every(
+          f => f.has_field_values === "search" || f.has_field_values === "list",
+        )))
```

Comment:
This condition is brutal. Any chance we can simplify it? Is there any value in this right now?